source/backends/debian/debpkg.d

```diff
@@ -56,7 +58,16 @@ public:
     @property override const(string[string]) description () const { return desc; }
 
     override
-    @property string filename () const { return debFname; }
+    @property string filename () const {
+        if (debFname.isRemote) {
+            immutable path = buildPath (tmpDir, debFname.baseName);
+
+            downloadFile (debFname, path);
```

Comment:
`downloadFile` itself avoids re-downloading if the dest exists. Here I have the entire path, so it's easier to download directly; `downloadIfNecessary` is easier to use if you're constructing the path yourself from a base and suffix.

(That reminds me that `downloadIfNecessary`'s doc comment needs expanding...)